
# HG changeset patch
# User Magnus Melin <mkmelin+mozilla@iki.fi>
# Date 1698443620 -7200
# Node ID bbcdfae7ff3a782095e34a22e521b650781f64b0
# Parent  380fd9a2a7f2fd6968a7a2652ed87376f8c2cf2a
Bug 1841547 - Don't allow trying to publish calendar to incorrect URLs. r=john.bieling

Make the publish button disabled if it's not an http url.

Differential Revision: https://phabricator.services.mozilla.com/D191619

diff --git a/calendar/base/content/dialogs/publishDialog.js b/calendar/base/content/dialogs/publishDialog.js
--- a/calendar/base/content/dialogs/publishDialog.js
+++ b/calendar/base/content/dialogs/publishDialog.js
@@ -12,34 +12,39 @@ window.addEventListener("DOMContentLoade
  */
 function loadCalendarPublishDialog() {
   let args = window.arguments[0];
 
   gOnOkFunction = args.onOk;
 
   if (args.publishObject) {
     gPublishObject = args.publishObject;
-    if (args.publishObject.remotePath) {
+    if (
+      args.publishObject.remotePath &&
+      /^(https?|webcals?):\/\//.test(args.publishObject.remotePath)
+    ) {
       document.getElementById("publish-remotePath-textbox").value = args.publishObject.remotePath;
     }
   } else {
     gPublishObject = {};
   }
 
   checkURLField();
 
   let firstFocus = document.getElementById("publish-remotePath-textbox");
   firstFocus.focus();
 }
 
 /**
  * Called when the OK button is clicked.
  */
 function onOKCommand(event) {
-  gPublishObject.remotePath = document.getElementById("publish-remotePath-textbox").value;
+  gPublishObject.remotePath = document
+    .getElementById("publish-remotePath-textbox")
+    .value.replace(/^webcal/, "http");
 
   // call caller's on OK function
   gOnOkFunction(gPublishObject, progressDialog);
   let dialog = document.querySelector("dialog");
   dialog.getButton("accept").setAttribute("label", dialog.getAttribute("buttonlabelaccept2"));
   event.preventDefault();
 }
 document.addEventListener("dialogaccept", onOKCommand, { once: true });
diff --git a/calendar/base/content/dialogs/publishDialog.xhtml b/calendar/base/content/dialogs/publishDialog.xhtml
--- a/calendar/base/content/dialogs/publishDialog.xhtml
+++ b/calendar/base/content/dialogs/publishDialog.xhtml
@@ -19,30 +19,32 @@
   xmlns:html="http://www.w3.org/1999/xhtml"
   windowtype="Calendar:PublishDialog"
   persist="screenX screenY"
   lightweightthemes="true"
   scrolling="false"
 >
   <head>
     <title>&calendar.publish.dialog.title;</title>
+    <link rel="localization" href="branding/brand.ftl" />
     <script defer="defer" src="chrome://messenger/content/dialogShadowDom.js"></script>
     <script defer="defer" src="chrome://calendar/content/publishDialog.js"></script>
   </head>
   <html:body xmlns="http://www.mozilla.org/keymaster/gatekeeper/there.is.only.xul">
     <dialog
       buttons="accept,cancel"
       buttonlabelaccept="&calendar.publish.publish.button;"
       buttonlabelaccept2="&calendar.publish.close.button;"
     >
       <html:div>
         <html:label for="publish-remotePath-textbox">&calendar.publish.url.label;</html:label>
         <html:input
           id="publish-remotePath-textbox"
           type="url"
+          pattern="(https?|webcals?)://.*"
           size="64"
           required="required"
           placeholder="https://www.example.com/webdav/calendar.ics"
           oninput="checkURLField()"
         />
       </html:div>
       <html:progress id="publish-progressmeter" value="0" max="100"></html:progress>
     </dialog>
